src/software/simulation/standalone_simulator_main.cpp

```diff
@@ -1,20 +1,106 @@
+#include "shared/constants.h"
+#include "shared/proto/tbots_software_msgs.pb.h"
+#include "software/constants.h"
 #include "software/gui/standalone_simulator/standalone_simulator_gui_wrapper.h"
 #include "software/logger/logger.h"
+#include "software/networking/threaded_proto_multicast_listener.h"
+#include "software/networking/threaded_proto_multicast_sender.h"
+#include "software/parameter/dynamic_parameters.h"
+#include "software/primitive/primitive.h"
 #include "software/simulation/threaded_simulator.h"
 #include "software/world/field.h"
 
-int main(int argc, char *argv[])
+/**
+ * Decodes a PrimitiveMsg into its primitive index and parameters
+ *
+ * @param msg The message to decode
+ *
+ * @return The primitive index and parameters for the given PrimitiveMsg
+ */
+std::pair<unsigned int, primitive_params_t> decodePrimitiveMsg(const PrimitiveMsg& msg)
+{
+    auto primitive_index = static_cast<unsigned int>(msg.prim_type());
+
+    primitive_params_t params;
+    params.params[0] = static_cast<uint16_t>(msg.parameter1());
+    params.params[1] = static_cast<uint16_t>(msg.parameter2());
+    params.params[2] = static_cast<uint16_t>(msg.parameter3());
+    params.params[3] = static_cast<uint16_t>(msg.parameter4());
+    params.extra     = static_cast<uint8_t>(msg.extra_bits());
+    params.slow      = msg.slow();
+
+    return std::make_pair(primitive_index, params);
+}
+
+int main(int argc, char* argv[])
```

Comment:
this function is getting too long, can we break it up?

Reply:
done with the StandlaoneSimualtor class